API.md

```diff
@@ -2871,6 +2871,28 @@ const handler = function (request, reply) {
 };
 ```
 
+If the handler returns a Promise then Hapi will register a `catch` handler on the promise object to catch unhandled promise rejections. The handler will `reply` with the error:
+
+```js
+const handler = function (request, reply) {
+
+  const badPromise = () => {
+
+    new Promise(() => {
+
+      setTimeout(() => throw new Error(), 1000);
```

Comment:
Maybe there's something I'm missing here, but unless there's some magic going on behind the scenes with [domains](https://nodejs.org/api/domain.html) this should result in resolving with `undefined` and triggering an `uncaughtException` on the process.